docs/_docs/constraints.md

```diff
@@ -0,0 +1,261 @@
+
+**Best practice:** One way to reduce that API load is via the Zapier platform dehydration feature mentioned earlier. By putting these secondary requests behind a dehydration pointer, Zapier will only make this request once, although it might see these same records again and again based on the Zap’s polling cycle.
+
+More on dehydration again [here](https://platform.zapier.com/cli_docs/docs#dehydration).
+
+## Reducing file requests to your API
+
+**Constraint:** Each time a Zap step requests a file from your API, it will be accessed and downloaded.
+
+**Best practice:** Much like data dehydration, you can implement a file request using [dehydration](https://platform.zapier.com/cli_docs/docs#zgeneratecallbackurl#file-dehydration), so again the file will only be accessed and downloaded when a later Zap step asks for it.
+
+To make this even more efficient, you can [stash the file](https://platform.zapier.com/cli_docs/docs#stashing-files) at Zapier. Rather than provide the file to the requesting step, Zapier will stash the file (under a dehydrated URL), so that only one request will ever be made for the file from your API.
+
+# Throttling
+
+There are a number of throttles that a Zapier user could encounter when using your integration, including those set by your API.
+
+## Throttling (your API)
+
+**Constraint:** Your API has request limits
+
+**What errors a user will see if constraint hit:**
+
+- If a trigger, user will receive an email with an error message about the trigger error
+- If an action, user will see an error in Zap history
+
+**Best practice:** You can add a specific 429 `Retry-After` header to your response, or add a timed delay in your error response. Instead of a user’s Zap erroring and halting, the request will be retried at the specified time. More on the retry [here](https://platform.zapier.com/cli_docs/docs#handling-throttled-requests). The user will see this message in Zap History instead of an error while the limit is still in place:
+![](https://cdn.zappy.app/0933736266259b11771a0eba0aff23ce.png)
+
+## Webhook throttles (Zapier)
+
+**Constraint:** Zapier’s current webhook limits are [here](https://zapier.com/help/troubleshoot/behavior/rate-limits-and-throttling-in-zapier#step-4). We will issue a 429 response when your integration exceeds these throttle limits.
+
+**What errors a user will see:**
+
+- If a trigger, user will receive an email with an error message about the trigger error
+- If an action, user will see error in Zap history
+
+**Best practice:** You should support a retry/back-off schedule to make sure the data is eventually received.
+
+## Polling trigger throttle (Zapier)
+
+**Constraint:** There is a default limit of 100 returned items/poll.
```

Comment:
```suggestion
**Constraint:** There is a default limit of 100 new items/poll after deduplication.
```

Reply:
I'm thinking we might actually want to link directly to the held runs docs here, as those writing may not yet have come across it.